js/meshopt_clusterizer.js

```diff
@@ -0,0 +1,380 @@
+		var indicesp = sbrk(indices.byteLength);
+		var verticesp = sbrk(vertex_positions.byteLength);
+
+		// copy input data to wasm memory
+		var heap = new Uint8Array(instance.exports.memory.buffer);
+		heap.set(bytes(indices), indicesp);
+		heap.set(bytes(vertex_positions), verticesp);
+
+		instance.exports.meshopt_computeClusterBounds(
+			resultp,
+			indicesp,
+			indices.length,
+			verticesp,
+			vertex_count,
+			vertex_positions_stride
+		);
+
+		// heap might (will?) have grown -> re-acquire
+		heap = new Uint8Array(instance.exports.memory.buffer);
+
+		var boundsBytes = heap.subarray(resultp, resultp + BOUNDS_SIZE);
+
+		var result = {
+			center: copyTypedArray(new Float32Array(boundsBytes.buffer, boundsBytes.byteOffset, 3)),
+			radius: new Float32Array(boundsBytes.buffer, boundsBytes.byteOffset + 3 * Float32Array.BYTES_PER_ELEMENT, 1)[0],
+			coneApex: copyTypedArray(new Float32Array(boundsBytes.buffer, boundsBytes.byteOffset + 4 * Float32Array.BYTES_PER_ELEMENT, 3)),
+			coneAxis: copyTypedArray(new Float32Array(boundsBytes.buffer, boundsBytes.byteOffset + 7 * Float32Array.BYTES_PER_ELEMENT, 3)),
+			coneCutoff: new Float32Array(boundsBytes.buffer, boundsBytes.byteOffset + 10 * Float32Array.BYTES_PER_ELEMENT, 1)[0],
+			coneAxisS8: copyTypedArray(new Int8Array(boundsBytes.buffer, boundsBytes.byteOffset + 11 * Float32Array.BYTES_PER_ELEMENT, 3)),
+			coneCutoffS8: new Int8Array(boundsBytes.buffer, boundsBytes.byteOffset + 47, 1)[0],
+		};
+
+		// reset memory
+		sbrk(resultp - sbrk(0));
+
+		return result;
+	}
+
+	return {
+		ready: ready,
+		supported: true,
+		buildMeshlets: function (indices,
+								 vertex_positions,
+								 vertex_positions_stride,
+								 max_vertices,
+								 max_triangles,
+								 cone_weight,
+								 index_byte_size) {
+			assert(vertex_positions.length % vertex_positions_stride === 0);
+			var vertex_count = vertex_positions.length / vertex_positions_stride;
+			cone_weight = cone_weight || 0.0;
+			if (max_vertices > 255 || max_vertices < 0) {
+				throw new Error('Invalid max vertices');
+			}
+			if (max_triangles > 512) {
+				throw new Error('Invalid max triangles');
+			}
+			if (max_triangles % 4 !== 0) {
+				throw new Error('Invalid max triangles');
+			}
+
+			var meshlets = buildMeshlets(
+				index32(indices, index_byte_size),
+				vertex_count,
+				max_vertices,
+				max_triangles,
+				cone_weight,
+				vertex_positions,
+				vertex_positions_stride * Float32Array.BYTES_PER_ELEMENT
+			);
+
+			var result = [];
+			for (var i = 0; i < meshlets.meshletCount; ++i) {
+				var m = meshlets.getMeshlet(i);
+				optimize(m.vertices, m.triangles);
+				result.push(m);
+			}
+
+			return result;
+		},
+		computeClusterBounds: function (indices,
+										vertex_positions,
+										vertex_positions_stride,
+										index_byte_size) {
+			assert(vertex_positions.length % vertex_positions_stride === 0);
+			var ind32 = index32(indices, index_byte_size);
+			if (ind32.length / 3 > 512) {
```

Comment:
These two checks should probably use assert for consistency.

Reply:
done: [894160e](https://github.com/zeux/meshoptimizer/pull/738/commits/894160e14564daded8e460ab580b22ea7034da27)